### ISSUE TITLE
Pasted markdown lists are not displayed in CKEditor

### ISSUE THREAD

**Issue by hamza221** [nextcloud/mail#9858]
### Steps to reproduce

1. Create a list in an external markdown editor
2. Copy it
3. Paste it in Composer

### Expected behavior

Display the list correctly 

### Actual behavior

displayed as normal text no indication (*,- or 1.) that it's a list

### PHP engine version

None

### Web server

None

### Database

None

**Comment by ChristophWurst** [comment]
Dup of https://github.com/nextcloud/mail/issues/9608

**Comment by hamza221** [comment]
fixed in https://github.com/nextcloud/mail/pull/9848

**Comment by alexanderdd** [comment]
@hamza221 is this already deployed on c.nc.com? Cause I just pasted a copied markdown text and it did not copy the list formatting.

**Comment by alexanderdd** [comment]
I just copied a bullet list from an md document (app "Text") to an email on c.nc.com -> the bullet formatting was dropped. 

Firefox 129. Cleared cache before.

**Comment by ChristophWurst** [comment]
Could this be a Text issue? Copying lists from Text to Talk is also kind of broken. Some list handles are lost, others are preserved.

**Comment by ChristophWurst** [comment]
@alexanderdd copying lists from Text to the native Gnome text editor doesn't work either. Top level items lose their handle. Sub items remain lists.

Pasting a markdown list into CKEditor keeps the markdown format. I'd says that is somewhat expected because you don't copy a formatted string but "source code". We could consider detecting markdown and doing a conversion, as a feature.

**Comment by alexanderdd** [comment]
Sorry I don't understand what info you need.
As a user, I expect that I can copy a bullet list from NC Text to NC Mail and formatting will be preserved. Not sure what needs to be done for this to work.

**Comment by ChristophWurst** [comment]
Thank you for your feedback

**Comment by i2h3** [comment]
I still have this issue all the time when copying work reports from the Nextcloud Notes web editor into the Nextcloud Mail compose editor (Safari Version 18.1.1 (20619.2.8.11.12)).

**Comment by ChristophWurst** [comment]
Thank you for the feedback @i2h3. When copying from Notes or Text the clipboard contains broken markdown. This markdown is taken as-is by our editor.
For Notes/Text contents to be copy-pastable to Mail it would need the markdown text to be fully copyable, an automatic detection of markdown, and a conversion from markdown to HTML.
This is what happens when you copy from Notes/Text to a local text editor:
![Image](https://github.com/user-attachments/assets/532b036a-d8c8-4bc9-8b09-8a053cadc1c8)

As a counter example: go to https://www.w3schools.com/html/html_lists.asp and copy a list. Paste it into Mail. The bullet points are kept, because your clipboard will contain HTML. There is no conversion necessary.

tl;dr: the problem is the content copied from Notes/Text

**Comment by alexanderdd** [comment]
@ChristophWurst 
> tl;dr: the problem is the content copied from Notes/Text

So what is needed is a bug report in NC Text, correct? Can you do it? Otherwise I can do it.

**Comment by ChristophWurst** [comment]
That is correct, @alexanderdd! I checked this morning and could not find any existing reports about this. Thanks for taking the lead on this.

**Comment by i2h3** [comment]
Well, that is funny… I started writing the issue for Nextcloud Text myself and while trying to reproduce the issue for exact descriptions of the steps involved it works. Copy and paste works from Nextcloud Text. 😕 Also from Nextcloud Notes. 😵‍💫 Both into macOS' TextEdit app, both plain and formatted text. But not into the editor of Nextcloud Mail. 🤔

**Comment by ChristophWurst** [comment]
@i2h3 well observed.

We have to study where the formatting is dropped. https://ckeditor.com/ckeditor-5/demo/feature-rich/ does work with a copy-paste from Nextcloud Text (when using heading+list)

**Comment by juliusknorr** [comment]
I actually think this is a mail issue. Testing with https://ckeditor.com/ckeditor-5/demo/feature-rich/ I can confirm that copying form text works and also inspecting the clipboard content from text with https://evercoder.github.io/clipboard-inspector/ we properly pass over HTML and markdown including the list formatting.

**Comment by alexanderdd** [comment]
Just pasted some unnumbered lists from NC Text into Thunderbird - works fine. 

So I reopen this. (also because of Julius' comment)

Should I close https://github.com/nextcloud/text/issues/6708 ?

**Comment by ChristophWurst** [comment]
> We have to study where the formatting is dropped. https://ckeditor.com/ckeditor-5/demo/feature-rich/ does work with a copy-paste from Nextcloud Text (when using heading+list)

Yes

**Comment by alexanderdd** [comment]
this is still a problem in current cloud.nextcloud.com (cannot check mail app version).

@ChristophWurst can you bring this up in a planning meeting to see if it can be scheduled?
